crates/pop-parachains/src/call/metadata/action.rs

```diff
@@ -0,0 +1,208 @@
+		props(Pallet = "Balances")
+	)]
+	Transfer,
+	/// Create an asset.
+	#[strum(
+		serialize = "create",
+		message = "create",
+		detailed_message = "Create an asset",
+		props(Pallet = "Assets")
+	)]
+	CreateAsset,
+	/// Mint an asset.
+	#[strum(
+		serialize = "mint",
+		message = "mint",
+		detailed_message = "Mint an asset",
+		props(Pallet = "Assets")
+	)]
+	MintAsset,
+	/// Create an NFT collection.
+	#[strum(
+		serialize = "create_nft",
+		message = "create",
+		detailed_message = "Create an NFT collection",
+		props(Pallet = "Nfts")
+	)]
+	CreateCollection,
+	/// Mint an NFT.
+	#[strum(
+		serialize = "mint_nft",
+		message = "mint",
+		detailed_message = "Mint an NFT",
+		props(Pallet = "Nfts")
+	)]
+	MintNFT,
+	/// Purchase on-demand coretime.
+	#[strum(
+		serialize = "place_order_allow_death",
+		message = "place_order_allow_death",
+		detailed_message = "Purchase on-demand coretime",
+		props(Pallet = "OnDemand")
+	)]
+	PurchaseOnDemandCoretime,
+	/// Reserve a parachain ID.
+	#[strum(
+		serialize = "reserve",
+		message = "reserve",
+		detailed_message = "Reserve a parachain ID",
+		props(Pallet = "Registrar")
+	)]
+	Reserve,
+	/// Register a parachain ID with genesis state and code.
+	#[strum(
+		serialize = "register",
+		message = "register",
+		detailed_message = "Register a parachain ID with genesis state and code",
+		props(Pallet = "Registrar")
+	)]
+	Register,
+}
+
+impl Action {
+	/// Get the description of the action.
+	pub fn description(&self) -> &str {
+		self.get_detailed_message().unwrap_or_default()
+	}
+
+	/// Get the extrinsic name corresponding to the action.
+	pub fn extrinsic_name(&self) -> &str {
+		self.get_message().unwrap_or_default()
+	}
+
+	/// Get the associated pallet name for the action.
+	pub fn pallet_name(&self) -> &str {
+		self.get_str("Pallet").unwrap_or_default()
+	}
+}
+
+/// Fetch the list of supported actions based on available pallets.
+///
+/// # Arguments
+///
+/// * `pallets`: List of pallets availables in the chain.
+pub async fn supported_actions(pallets: &[Pallet]) -> Vec<Action> {
+	let mut actions = Vec::new();
+	for action in Action::VARIANTS.iter() {
+		if find_extrinsic_by_name(pallets, action.pallet_name(), action.extrinsic_name())
+			.await
+			.is_ok()
+		{
+			actions.push(action.clone());
+		}
+	}
+	actions
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate::{parse_chain_metadata, set_up_client};
+	use anyhow::Result;
+	use std::collections::HashMap;
+
+	#[test]
+	fn action_descriptions_are_correct() {
+		let descriptions = HashMap::from([
+			(Action::CreateAsset, "Create an asset"),
+			(Action::MintAsset, "Mint an asset"),
+			(Action::CreateCollection, "Create an NFT collection"),
+			(Action::MintNFT, "Mint an NFT"),
+			(Action::PurchaseOnDemandCoretime, "Purchase on-demand coretime"),
+			(Action::Transfer, "Transfer balance"),
+			(Action::Register, "Register a parachain ID with genesis state and code"),
+			(Action::Reserve, "Reserve a parachain ID"),
```

Comment:
We are testing `Action::VARIANTS.iter()`, see [here](https://github.com/r0gue-io/pop-cli/blob/feat-call-ui-extrinsics-limited/crates/pop-parachains/src/call/metadata/action.rs#L142), and comparing it with this values. Changing it would make the test irrelevant.